tests/tracking/test_fluent.py

```diff
@@ -401,29 +402,107 @@ def test_search_runs_no_arguments():
     # When no experiment ID is specified,
     # it should try to get the implicit one or create a new experiment
     mock_experiment_id = mock.Mock()
-    experiment_id_patch = mock.patch(
-        "mlflow.tracking.fluent._get_experiment_id", return_value=mock_experiment_id
-    )
-    with experiment_id_patch, mock.patch.object(MlflowClient, "search_runs", return_value=[]):
+    experiment_id_patch = mock.patch("mlflow.tracking.fluent._get_experiment_id",
+                                     return_value=mock_experiment_id)
+    get_paginated_runs_patch = mock.patch('mlflow.tracking.fluent._get_paginated_runs',
+                                          return_value=[])
+    with experiment_id_patch, get_paginated_runs_patch:
         pdf = search_runs()
-        MlflowClient.search_runs.assert_called_once_with(
+        mlflow.tracking.fluent._get_paginated_runs.assert_called_once_with(
             mock_experiment_id, '', ViewType.ACTIVE_ONLY, SEARCH_MAX_RESULTS_PANDAS, None
         )
 
 
-def test_search_runs_with_arguments():
-    mock_experiment_ids = mock.Mock()
-    mock_filter_string = mock.Mock()
-    mock_view_type = mock.Mock()
-    mock_max_results = mock.Mock()
-    mock_order_by = mock.Mock()
-    with mock.patch.object(MlflowClient, "search_runs", return_value=[]):
-        pdf = search_runs(mock_experiment_ids, mock_filter_string, mock_view_type,
-                          mock_max_results, mock_order_by)
-        MlflowClient.search_runs.assert_called_once_with(
-            mock_experiment_ids,
-            mock_filter_string,
-            mock_view_type,
-            mock_max_results,
-            mock_order_by
-        )
+def test_get_paginated_runs_lt_maxresults_blanktoken():
+    # num runs is less than max_results, only one page
+    runs = [create_run() for i in range(5)]
+    tokenized_runs = PagedList(runs, "")
+    max_results = 50
+    with mock.patch("mlflow.tracking.fluent.NUM_RUNS_PER_PAGE_PANDAS", 10):
+        with mock.patch.object(MlflowClient, "search_runs", return_value=tokenized_runs):
+            paginated_runs = _get_paginated_runs([], "", ViewType.ACTIVE_ONLY, max_results, None)
+            MlflowClient.search_runs.assert_called_once()
+            assert len(paginated_runs) == 5
+
+
+def test_get_paginated_runs_lt_maxresults_notoken():
+    # num runs is less than max_results, only one page
+    # the list returned is not a PagedList, no token attribute
```

Comment:
Why is this possible actually?

Reply:
I think this might not be the case anymore now that tokens are implemented in OSS stores. Before the interface said that there would be a "token attribute" if the tracking server supported pagination. But at this version of MLflow, they should all support it now.